app/vmagent/remotewrite/client.go

```diff
@@ -459,6 +458,35 @@ again:
 		_ = resp.Body.Close()
 		c.packetsDropped.Inc()
 		return true
+	} else if statusCode == http.StatusUnsupportedMediaType {
+		if c.useVMProto.Swap(false) {
+			logger.Infof("received unsupported media type from remote storage at %q. Downgrading protocol from VictoriaMetrics to Prometheus remote write for all future requests. "+
+				"See https://docs.victoriametrics.com/vmagent/#victoriametrics-remote-write-protocol", c.sanitizedURL)
+		}
+
+		if encoding.IsZstd(block) {
+			logger.Infof("received unsupported media type from remote storage at %q. Re-packing the block to Prometheus remote write and retrying."+
+				"See https://docs.victoriametrics.com/vmagent/#victoriametrics-remote-write-protocol", c.sanitizedURL)
+
+			block, err = repackBlockFromZstdToSnappy(block)
+			if err != nil {
+				logger.Errorf("cannot re-pack block with size %d bytes from Zstd to Snappy for %q: %s; the block will be dropped",
+					len(block), c.sanitizedURL, err)
+				_ = resp.Body.Close()
+				c.packetsDropped.Inc()
+				return true
+			}
+
+			c.retriesCount.Inc()
+			goto again
+		}
+
+		logger.Errorf("received unsupported media type from remote storage at %q. The block is already in Snappy format and cannot be re-packed, so it will be dropped." +
```

Comment:
It would be great logging the response body contents here, since it may contain information, which help resolving the issue more quickly from the users' perspective.